src/axom/core/detail/FlatMapOps.hpp

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2017-2025, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef Axom_Core_Detail_FlatMapOps_Hpp
+#define Axom_Core_Detail_FlatMapOps_Hpp
+
+#include "axom/core/detail/FlatTable.hpp"
+
+namespace axom
+{
+namespace detail
+{
+namespace flat_map
+{
+
+inline void setSentinel(axom::ArrayView<GroupBucket> metadata)
+{
+#if defined(AXOM_USE_UMPIRE) && defined(AXOM_USE_CUDA)
```

Comment:
Does/Will this also support hip? Or does hip not need extra handling?
If it's the latter, please add a comment.

Reply:
Yes, it's the latter case since with HIP we can access device memory directly from the host. I'll add a comment.

Reply:
That's true on some Hip platforms, but not all.

Reply:
Added a comment.